oteljava/common/src/main/scala/org/typelevel/otel4s/oteljava/AttributeConverters.scala

```diff
@@ -0,0 +1,196 @@
+    def toScala: Attributes = Explicit.toScala(attributes)
+  }
+
+  private object Explicit {
+    def toJava(key: AttributeKey[_]): JAttributeKey[_] =
+      key.`type` match {
+        case AttributeType.String     => JAttributeKey.stringKey(key.name)
+        case AttributeType.Boolean    => JAttributeKey.booleanKey(key.name)
+        case AttributeType.Long       => JAttributeKey.longKey(key.name)
+        case AttributeType.Double     => JAttributeKey.doubleKey(key.name)
+        case AttributeType.StringList => JAttributeKey.stringArrayKey(key.name)
+        case AttributeType.BooleanList =>
+          JAttributeKey.booleanArrayKey(key.name)
+        case AttributeType.LongList   => JAttributeKey.longArrayKey(key.name)
+        case AttributeType.DoubleList => JAttributeKey.doubleArrayKey(key.name)
+      }
+
+    final def toJava(attributes: Attributes): JAttributes =
+      toJavaAttributes(attributes)
+
+    def toJavaAttributes(
+        attributes: immutable.Iterable[Attribute[_]]
+    ): JAttributes = {
+      val builder = JAttributes.builder
+
+      attributes.foreach { case Attribute(key, value) =>
+        key.`type` match {
+          case AttributeType.String =>
+            builder.put(key.name, value.asInstanceOf[String])
+          case AttributeType.Boolean =>
+            builder.put(key.name, value.asInstanceOf[Boolean])
+          case AttributeType.Long =>
+            builder.put(key.name, value.asInstanceOf[Long])
+          case AttributeType.Double =>
+            builder.put(key.name, value.asInstanceOf[Double])
+          case AttributeType.StringList =>
+            builder.put(key.name, value.asInstanceOf[List[String]]: _*)
+          case AttributeType.BooleanList =>
+            builder.put(key.name, value.asInstanceOf[List[Boolean]]: _*)
+          case AttributeType.LongList =>
+            builder.put(key.name, value.asInstanceOf[List[Long]]: _*)
+          case AttributeType.DoubleList =>
+            builder.put(key.name, value.asInstanceOf[List[Double]]: _*)
+        }
+      }
+
+      builder.build()
+    }
+
+    def toScala(key: JAttributeKey[_]): AttributeKey[_] =
+      key.getType match {
+        case JAttributeType.STRING       => AttributeKey.string(key.getKey)
+        case JAttributeType.BOOLEAN      => AttributeKey.boolean(key.getKey)
+        case JAttributeType.LONG         => AttributeKey.long(key.getKey)
+        case JAttributeType.DOUBLE       => AttributeKey.double(key.getKey)
+        case JAttributeType.STRING_ARRAY => AttributeKey.stringList(key.getKey)
+        case JAttributeType.BOOLEAN_ARRAY =>
+          AttributeKey.booleanList(key.getKey)
+        case JAttributeType.LONG_ARRAY   => AttributeKey.longList(key.getKey)
+        case JAttributeType.DOUBLE_ARRAY => AttributeKey.doubleList(key.getKey)
+      }
+
+    def toScala(attributes: JAttributes): Attributes = {
+      val builder = Attributes.newBuilder
+
+      val f: (JAttributeKey[_], Any) => Unit =
```

Comment:
We can even move the body of `f` to `forEach`:
```scala
attributes.forEach { (key, value) =>
  key.getType match {
    ...
  }
}
```

Similarly to https://github.com/typelevel/otel4s/pull/544/files#diff-5108665af842afe9ec9bee56f41e053c250dee57219b287dbafba8a58c614e92R119-R138.

Reply:
There is also a similar utility method in the oteljava-testkit: https://github.com/typelevel/otel4s/blob/main/oteljava/common-testkit/src/main/scala/org/typelevel/otel4s/oteljava/testkit/Conversions.scala#L33.

Can we replace `testkit.Conversions.fromJAttributes` with `import org.typelevel.otel4s.oteljava.AttributeConverters._` across the codebase? There are around 3 total usages, if I'm not mistaken.

Reply:
> We can move the body of `f` to `forEach`

nope, the SAM type inference doesn't work because... I don't remember why actually. but it didn't.

> Can we replace `testkit.Conversions.fromJAttributes` with `import org.typelevel.otel4s.oteljava.AttributeConverters._` across the testkit codebase? There are around 3 total usages, if I'm not mistaken.

absolutely! I started this before it existed I believe, and I meant to go back and do that but forgot

Reply:
> nope, the SAM type inference doesn't work because... I don't remember why actually. but it didn't.

and now it does, for some reason 🙃